### PR TITLE
Fix naked unwraps in light-client

### LINKED ISSUE #1045: Fix naked unwraps in `light-client` crate
Add `#![warn(clippy::unwrap_used)]` clippy lint to the `light-client/src/lib.rs`  file and remove all unwraps from the library.

### PR DESCRIPTION
**Pull request by abastola0**
Closes #1045

### CODE REVIEW

**Comment by abastola0** [comment]
> lmk if you require any error handling for the unwraps those were removed.

I guess we need to use them either way. What would you recommend I use here?

**Comment by sseyren** [comment]
Hi @abastola0, I'm also interested with this issue and created #1059. I didn't try to forcefully remove all `unwrap`s, instead, I turned them into `expect`s with appropriate info messages. If we can all agree on this, we can go that way.

**Comment by abastola0** [comment]
> Hi @abastola0, I'm also interested with this issue and created #1059. I didn't try to forcefully remove all `unwrap`s, instead, I turned them into `expect`s with appropriate info messages. If we can all agree on this, we can go that way. 

Sure takeover.

File: light-client/README.md [hunk 1]
```diff
@@ -51,15 +51,15 @@ async fn main() -> Result<()> {
     builder = builder.load_external_fallback();
 
     // Build the client
-    let mut client: Client<FileDB> = builder.build().unwrap();
+    let mut client: Client<FileDB> = builder.build();
```
Comment (njgheorghita) [inline light-client/README.md:54]: It's fine to leave naked unwraps inside of the `README.md`

File: light-client/src/config/checkpoints.rs [hunk 2]
```diff
@@ -293,39 +293,26 @@ mod test {
 
     #[tokio::test]
     async fn test_construct_checkpoints() {
-        let cf = config::checkpoints::CheckpointFallback::new()
-            .build()
-            .await
-            .unwrap();
+        let cf = config::checkpoints::CheckpointFallback::new().build().await;
```
Comment (njgheorghita) [inline light-client/src/config/checkpoints.rs:296]: We're also fine with leaving naked unwraps inside of tests. It's only naked unwraps inside of the actual code that is a concern. So, I'd leave all unwraps inside of the tests as they were, and you can use [this](https://github.com/ethereum/trin/blob/master/portalnet/src/overlay_service.rs#L2651) to ask that clippy ignore the test cases

**Review by njgheorghita — COMMENTED** [review]
Left a few comments that need addressing. Also note that you will need to add a line of code to require that clippy checks for this case inside of `light-client/src/lib.rs` - like we do [here](https://github.com/ethereum/trin/blob/master/ethportal-api/src/lib.rs#L4)

**Pull request closed without merging**